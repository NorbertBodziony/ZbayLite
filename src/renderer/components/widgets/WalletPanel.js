import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'

import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

const styles = {
}

export const WalletPanel = ({ classes }) => (
  <Grid container direction='column' className={classes.root}>
    <Typography>
      Available
    </Typography>
    <Grid item>
      <Grid container direction='row' justify='space-between'>
        <Typography>
          $2,366 USD
        </Typography>
        <Typography>
          z 33.5830004
        </Typography>
      </Grid>
    </Grid>
  </Grid>
)

WalletPanel.propTypes = {
  classes: PropTypes.object.isRequired
}

export default R.compose(
  React.memo,
  withStyles(styles)
)(WalletPanel)
