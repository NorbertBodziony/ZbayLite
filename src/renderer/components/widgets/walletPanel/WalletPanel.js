import React, { useState } from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'

import Grid from '@material-ui/core/Grid'
import { withStyles } from '@material-ui/core/styles'

import WalletPanelActions from '../../../containers/widgets/walletPanel/WalletPanelActions'
import UsdBalance from '../../../containers/widgets/walletPanel/UsdBalance'
import ZecBalance from '../../../containers/widgets/walletPanel/ZecBalance'
import TopUpModal from '../../../containers/widgets/walletPanel/TopUpModal'

const styles = theme => ({
  root: {
    paddingLeft: 2 * theme.spacing.unit,
    paddingTop: 2.5 * theme.spacing.unit,
    paddingRight: 4 * theme.spacing.unit,
    paddingBottom: theme.spacing.unit
  },
  actions: {
    marginTop: 2 * theme.spacing.unit
  },
  zec: {
    paddingTop: 2 * theme.spacing.unit
  }
})

export const WalletPanel = ({ classes, handleReceive }) => {
  const [topUpOpen, setTopUpOpen] = useState(false)

  return (
    <React.Fragment>
      <Grid item container direction='column' className={classes.root}>
        <Grid item>
          <Grid container direction='row' justify='space-between' alignItems='center' >
            <Grid item>
              <UsdBalance />
            </Grid>
            <Grid item className={classes.zec}>
              <ZecBalance />
            </Grid>
          </Grid>
        </Grid>
        <Grid item className={classes.actions}>
          <WalletPanelActions onReceive={() => setTopUpOpen(true)} />
        </Grid>
      </Grid>
      <TopUpModal open={topUpOpen} handleClose={() => setTopUpOpen(false)} />
    </React.Fragment>
  )
}

WalletPanel.propTypes = {
  classes: PropTypes.object.isRequired
}

export default R.compose(
  React.memo,
  withStyles(styles)
)(WalletPanel)
