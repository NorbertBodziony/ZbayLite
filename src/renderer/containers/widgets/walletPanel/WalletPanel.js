import React from 'react'
import PropTypes from 'prop-types'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as R from 'ramda'

import WalletPanelComponent from '../../../components/widgets/walletPanel/WalletPanel'
import ratesHandlers from '../../../store/handlers/rates'
import { useInterval } from '../../hooks'
import { actions } from '../../../store/handlers/app'
import { actionCreators } from '../../../store/handlers/modals'

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      fetchPrices: ratesHandlers.epics.fetchPrices,
      setModalTab: () => actions.setModalTab('invite'),
      openInvitationModal: actionCreators.openModal('accountSettingsModal')
    },
    dispatch
  )

export const WalletPanel = ({ fetchPrices, ...props }) => {
  useInterval(fetchPrices, 15000)
  return <WalletPanelComponent {...props} />
}

WalletPanel.propTypes = {
  fetchPrices: PropTypes.func.isRequired
}

export default R.compose(
  React.memo,
  connect(
    null,
    mapDispatchToProps
  )
)(WalletPanel)
