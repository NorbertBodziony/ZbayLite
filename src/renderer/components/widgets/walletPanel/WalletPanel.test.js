/* eslint import/first: 0 */
import React from 'react'
import { shallow } from 'enzyme'

import { mockClasses } from '../../../../shared/testing/mocks'
import { WalletPanel } from './WalletPanel'

describe('WalletPanelActions', () => {
  it('renders component', () => {
    const result = shallow(
      <WalletPanel
        classes={mockClasses}
        handleInvitation={jest.fn()}
      />
    )
    expect(result).toMatchSnapshot()
  })
})
