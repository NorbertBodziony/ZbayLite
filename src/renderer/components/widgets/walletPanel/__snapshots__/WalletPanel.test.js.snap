// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletPanelActions renders component 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="root-mock"
    container={true}
    direction="row"
    item={true}
    justify="space-between"
  >
    <Memo(Connect(WithStyles(ZcashBalance))) />
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <Connect(WithStyles(undefined)) />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(undefined)
    action={[MockFunction]}
    text="Invite friends"
  />
</Fragment>
`;
